mono/metadata/boehm-gc.c

```diff
@@ -1129,7 +1130,7 @@ mono_gc_get_managed_allocator (MonoClass *klass, gboolean for_box, gboolean know
 		return NULL;
 	if (mono_class_has_finalizer (klass) || mono_class_is_marshalbyref (klass))
 		return NULL;
-	if (mono_profiler_get_events () & (MONO_PROFILE_ALLOCATIONS | MONO_PROFILE_STATISTICAL))
+	if (G_UNLIKELY (mono_profiler_allocations_enabled ()))
```

Comment:
I only implemented it for SGen. Boehm will just continue to use the native allocation functions as it always has. Well, unless there's some compelling reason why we should bother with adding new features to Boehm.